openmc/polynomial.py

```diff
@@ -75,7 +75,68 @@ def order(self):
     def __call__(self, r):
         import openmc.lib as lib
         if isinstance(r, Iterable):
-            return [np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r_i))
+            return [np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r_i / self.radius))
                     for r_i in r]
         else:
-            return np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r))
+            return np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r / self.radius))
+
+
+class Zernike(Polynomial):
+    """Create Zernike polynomials given coefficients and domain.
+    The Zernike polynomials are defined as in
+    :class:`ZernikeFilter`.
+    Parameters
+    ----------
+    coef : Iterable of float
+        A list of coefficients of each term in radial only Zernike polynomials
+    radius : float
+        Domain of Zernike polynomials to be applied on. Default is 1.
+    r : float
+        Position to be evaluated, normalized on radius [0,1]
+    theta : float 
+        Azimuthal to be evaluated, normalized on :math:`[0, 2*\pi]`
+        
+    Attributes
+    ----------
+    order : int
+        The maximum (even) order of Zernike polynomials.
+    radius : float
+        Domain of Zernike polynomials to be applied on. Default is 1.
+    theta : float 
+        Azimuthal of Zernike polynomial to be applied on. Default is 0.    
+    norm_coef : iterable of float
+        The list of coefficients of each term in the polynomials after
+        normailization.
+    """
+    def __init__(self, coef, radius=1):
+        super().__init__(coef)
+        self._order = int(((np.sqrt(8 * len(self.coef) + 1)) - 3) / 2)
```

Comment:
`np.sqrt` should only be used if you are applying sqrt to an array. For a scalar value, use `math.sqrt`.
```suggestion
        self._order = int((math.sqrt(8 * len(self.coef) + 1) - 3) / 2)
```
Also, this looks like it comes out of nowhere unless you happen to know that N = (order+1)(order+2)/2. I would add a comment that mentions this fact and that you're just solving a quadratic equation for the order.